app/services/transfers/service.go

```diff
@@ -0,0 +1,230 @@
+	if !isSufficient {
+		bs.logger.Errorf("Fee validation for TX ID [%s] failed. Provided tx reimbursement fee is invalid/insufficient. Error [%s]", tm.TransactionId, err)
+		err = bs.transactionRepository.UpdateStatusInsufficientFee(tm.TransactionId)
+		if err != nil {
+			bs.logger.Errorf("Failed to update status to [%s] of transaction with TransactionID [%s]. Error [%s].", transaction.StatusInsufficientFee, tm.TransactionId, err)
+			return err
+		}
+		bs.logger.Debugf("TX with ID [%s] was updated to [%s]. Provided fee [%s]", tm.TransactionId, transaction.StatusInsufficientFee, tm.Fee)
+		return err
+	}
+	return nil
+}
+
+// ProcessTransfer processes the transfer message by signing the required
+// authorisation signature submitting it into the required HCS Topic
+func (bs *Service) ProcessTransfer(tm encoding.TransferMessage) error {
+	authMsgHash, err := auth_message.EncodeBytesFrom(tm.TransactionId, tm.EthAddress, tm.Amount, tm.Fee)
+	if err != nil {
+		bs.logger.Errorf("Failed to encode the authorisation signature for TX ID [%s]. Error: %s", tm.TransactionId, err)
+		return err
+	}
+	signatureBytes, err := bs.ethSigner.Sign(authMsgHash)
+	if err != nil {
+		bs.logger.Errorf("Failed to sign the authorisation signature for TX ID [%s]. Error: %s", tm.TransactionId, err)
+		return err
+	}
+	signature := hex.EncodeToString(signatureBytes)
+	signatureMessage := encoding.NewSignatureMessage(
+		tm.TransactionId,
+		tm.EthAddress,
+		tm.Amount,
+		tm.Fee,
+		signature)
+
+	sigMsgBytes, err := signatureMessage.ToBytes()
+	if err != nil {
+		bs.logger.Error("Failed to encode Signature Message to bytes for TX [%s]. Error %s", err, tm.TransactionId)
+		return err
+	}
+	messageTxId, err := bs.hederaNode.SubmitTopicConsensusMessage(
+		bs.topicID,
+		sigMsgBytes)
+	if err != nil {
+		bs.logger.Errorf("Failed to submit Signature Message to Topic for TX [%s]. Error: %s", tm.TransactionId, err)
+		return err
+	}
+
+	// Update Transaction Record
+	tx, err := bs.transactionRepository.GetByTransactionId(tm.TransactionId)
+	if err != nil {
+		bs.logger.Errorf("Failed to get TX [%s] from DB", tm.TransactionId)
+		return err
+	}
+
+	tx.Signature = signature
+	tx.SignatureMsgTxId = messageTxId.String()
+	tx.Status = transaction.StatusInProgress
+	tx.SignatureMsgStatus = transaction.StatusSignatureSubmitted
+	err = bs.transactionRepository.Save(tx)
+	if err != nil {
+		bs.logger.Errorf("Failed to update TX [%s]. Error [%s].", tm.TransactionId, err)
+		return err
+	}
+
+	// Attach update callbacks on Signature HCS Message
+	bs.logger.Infof("Submitted signature for TX ID [%s] on Topic [%s]", tm.TransactionId, bs.topicID)
+	onSuccessfulAuthMessage, onFailedAuthMessage := bs.authMessageSubmissionCallbacks(tm.TransactionId)
+	bs.mirrorNode.WaitForTransaction(messageTxId.String(), onSuccessfulAuthMessage, onFailedAuthMessage)
+	return nil
+}
+
+func (bs *Service) authMessageSubmissionCallbacks(txId string) (func(), func()) {
```

Comment:
Can we set names to return values, in case someone does not want to read the whole method to understand what is returned first and second?
`onSuccess, onRevert func()`